app/src/test/kotlin/compiler/e2e/TypeCheckingErrorsTest.kt

```diff
@@ -1,5 +1,760 @@
 package compiler.e2e
 
+import compiler.common.diagnostics.Diagnostic
+import compiler.e2e.E2eAsserter.assertErrorOfType
+import org.junit.Ignore
+import org.junit.Test
+
 class TypeCheckingErrorsTest {
-    // TODO: implement tests for type checking errors
+    fun assertInvalidTypeError(program: String) {
+        assertErrorOfType(program, Diagnostic.InvalidType::class)
+    }
+
+    fun assertConditionalMismatchError(program: String) {
+        assertErrorOfType(program, Diagnostic.ConditionalTypesMismatch::class)
+    }
+
+    fun assertUninitializedGlobalVariableError(program: String) {
+        assertErrorOfType(program, Diagnostic.UninitializedGlobalVariable::class)
+    }
+
+    fun assertConstantWithoutValueError(program: String) {
+        assertErrorOfType(program, Diagnostic.ConstantWithoutValue::class)
+    }
```

Comment:
Those functions can be private, as well as assert functions in `VariablePropertiesAnalysisError`